browser/src/Services/Explorer/ExplorerStore.ts

```diff
@@ -109,6 +179,105 @@ export const rootFolderReducer: Reducer<IFolderState> = (
     }
 }
 
+// Helper functions for Updating state ========================================================
+export const removePastedNode = (nodeArray: ExplorerNode[], id: string): ExplorerNode[] =>
+    nodeArray.filter(node => node.id !== id)
+
+export const removeUndoItem = (undoArray: RegisterAction[]): RegisterAction[] =>
+    undoArray.slice(0, undoArray.length - 1)
+
+const getOriginalNodeLocations = (file: ExplorerNode, pasteTarget: ExplorerNode) => {
+    const originalFolder = path.dirname(getPathForNode(file))
+
+    const targetDirectory =
+        pasteTarget.type === "file"
+            ? path.dirname(pasteTarget.filePath)
+            : getPathForNode(pasteTarget)
+
+    const fileOrFolderPath = getPathForNode(file)
+    const filename = path.basename(fileOrFolderPath)
+    return {
+        file: path.join(targetDirectory, filename),
+        folder: originalFolder,
+    }
+}
+
+// Do not add un-undoable action to the undo list
+export const shouldAddDeletion = (action: IDeleteSuccessAction) => (action.persist ? [action] : [])
+
+const getUpdatedPasteNode = (action: IPasteAction) =>
+    action.pasted.map(node => getOriginalNodeLocations(node, action.target)).map(n => n.file)
+
+const getUpdatedDeleteNode = (action: IDeleteSuccessAction) => [getPathForNode(action.target)]
+
+type Updates = IPasteAction | IDeleteSuccessAction | IUndoSuccessAction
+
+export const getUpdatedNode = (action: Updates, state?: IRegisterState): string[] => {
+    switch (action.type) {
+        case "PASTE":
+            return getUpdatedPasteNode(action)
+        case "DELETE_SUCCESS":
+            return getUpdatedDeleteNode(action)
+        case "UNDO_SUCCESS":
+            const lastAction = tail(state.undo)
+            return lastAction.type === "DELETE_SUCCESS"
+                ? getUpdatedDeleteNode(lastAction)
+                : lastAction.pasted.map(node => getPathForNode(node))
+        default:
+            return []
+    }
+}
+
+// Yank, Paste Delete register =============================
+// The undo register is essentially a list of past actions
+// => [paste, delete, paste], when an action is carried out
+// it is added to the back of the stack when an undo is triggered
+// it is removed.
+// The most recently actioned node(s) path(s) are set to the value of
+// the updated field, this is used to animate updated fields.
+
+export const yankRegisterReducer: Reducer<IRegisterState> = (
+    state: IRegisterState = DefaultRegisterState,
+    action: ExplorerAction,
+) => {
+    switch (action.type) {
+        case "YANK":
+            return {
+                ...state,
+                yank: [...state.yank, action.target],
+            }
+        case "PASTE":
+            return {
+                ...state,
+                paste: action.target,
+                undo: [...state.undo, action],
```

Comment:
Cool! This reminds me of the undo example in the redux docs 😄  Tried it out and it works well, I like it!